speculos/mcu/ocr.py

```diff
@@ -150,63 +118,190 @@ def _get_font_map(font: bagl_font.Font) -> Mapping[Char, BitMapChar]:
+                struct_name = 'bagl_font_character'
+
+            for name in names:
+                # Add the fonts path (JSON files are in speculos/fonts)
+                json_name = os.path.join(fonts_path, name)
+                # Add api level information and file extension
+                json_name += f"-api-level-{OCR.api_level}.json"
+                if os.path.exists(json_name):
+                    self.get_json_font(json_name, struct_name)
+
+            if len(OCR.json_fonts) != 0:
+                OCR.legacy = False
+            else:
+                print("WARNING: didn't find any JSON font files => OCR will "\
+                      "not work properly!\n")
+
+    @staticmethod
+    def get_json_font(name, struct_name) -> Mapping[Char, BitMapChar]:
+        """
+        Read the JSON file and parse all character information
+        """
+        with open(name, "r") as json_file:
+            font_info = json.load(json_file, strict=False)
+            font_info = font_info[0]
+            # Deserialize bitmap
+            bitmap = base64.b64decode(font_info['bitmap'])
+            if not struct_name in font_info:
+                return
+            # Build BitMapChar
+            font_map = {}
+            for character in font_info[struct_name]:
+                char = character['char']
+                offset = character['bitmap_offset']
+                count = character['bitmap_byte_count']
+                # Add this entry in font_map
+                font_map[chr(char)] = BitMapChar(
+                    char,
+                    bytes(bitmap[offset:(offset + count)]),
+                )
+            OCR.json_fonts.append(font_map)
+
+    @staticmethod
+    def find_char_from_bitmap_legacy(bitmap: BitMap):
+        """
+        Find a character from a bitmap
+        >>> font = get_font(4)
+        >>> char = get_char(font, 'c')
+        >>> find_char_from_bitmap(char.bitmap)
+        'c'
+        """
+        all_values = []
+        for font in bagl_font.FONTS:
+            font_map = get_font_map(font)
             for character_value, bitmap_struct in font_map.items():
                 if bitmap_struct.bitmap.startswith(bitmap):
-                    # sometimes (but not always) the bitmap being passed is
-                    # shortened by one '\0' byte, not matching the exact bitmap
+                    # sometimes (but not always) the bitmap being passed is shortened
+                    # by one '\x00' byte, not matching the exact bitmap
                     # provided in the font. Hence the 'residual' computation
                     residual_bytes: bytes = bitmap_struct.bitmap[len(bitmap):]
                     if all(b == 0 for b in residual_bytes):
                         all_values.append(character_value)
+            if all_values:
+                char = max([x for x in all_values])
+                if char == "\x80":
+                    char = " "
+                return char
+        return None
```

Comment:
As the method `find_char_from_bitmap` is expected to return a value, it seemed normal to me to return `None` when no char was found.
(I know that doing nothing will return `None`, but its just clearer in my opinion)
But since you spent a lot of time doing the review, I'll do as you suggest :)